src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/ElasticsearchUserCaseAccessFilter.java

```diff
@@ -19,24 +23,30 @@
 public class ElasticsearchUserCaseAccessFilter implements CaseSearchFilter {
 
     private final CaseAccessService caseAccessService;
+    private final CaseDefinitionRepository caseDefinitionRepository;
 
     @Autowired
-    public ElasticsearchUserCaseAccessFilter(CaseAccessService caseAccessService) {
+    public ElasticsearchUserCaseAccessFilter(CaseAccessService caseAccessService,
+                                             @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                                             final CaseDefinitionRepository caseDefinitionRepository) {
         this.caseAccessService = caseAccessService;
+        this.caseDefinitionRepository = caseDefinitionRepository;
     }
 
     @Override
     public Optional<QueryBuilder> getFilter(String caseTypeId) {
         Instant start = Instant.now();
-        return getGrantedCaseReferencesForRestrictedRoles().map(caseReferences -> {
+        CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(caseTypeId);
+
```

Comment:
is it better to check for null values here in case caseTypeDefinition is null

Reply:
Well spotted.